src/otx/v2/adapters/torch/mmengine/mmdet/modules/datasets/dataset.py

```diff
@@ -0,0 +1,132 @@
+
+    """
+
+    class _DataInfoProxy:
+        """This class is intended to be a wrapper to use it in BaseDetDataset-derived class as `self.data_infos`.
+
+        Instead of using list `data_infos` as in BaseDetDataset, our implementation of dataset OTXDataset
+        uses this proxy class with overriden __len__ and __getitem__; this proxy class
+        forwards data access operations to otx_dataset and converts the dataset items to the view
+        convenient for mmdetection.
+
+        Args:
+            otx_dataset (DatasetEntity): DatasetEntity from dataset api
+            labels (List[LabelEntity]): List of LabelEntity
+        """
+
+        def __init__(self, otx_dataset: DatasetEntity, labels: list[LabelEntity]) -> None:
+            self.otx_dataset = otx_dataset
+            self.labels = labels
+            self.label_idx = {label.id: i for i, label in enumerate(labels)}
+
+        def __len__(self) -> int:
+            return len(self.otx_dataset)
+
+        def __getitem__(self, index: int) -> dict[str, Any]:
+            """Prepare a dict 'data_info' that is expected by the mmdet pipeline to handle images and annotations.
+
+            :return data_info: dictionary that contains the image and image metadata, as well as the labels of
+            the objects in the image
+            """
+            dataset = self.otx_dataset
+            item = dataset[index]
+            ignored_labels = np.array([self.label_idx[lbs.id] for lbs in item.ignored_labels])
+
+            height, width = item.height, item.width
+
+            return {
+                "dataset_item": item,
+                "width": width,
+                "height": height,
+                "index": index,
+                "ann_info": {"label_list": self.labels},
+                "ignored_labels": ignored_labels,
+                "bbox_fields": [],
+                "mask_fields": [],
+                "seg_fields": [],
+            }
+
+    def __init__(
+        self,
+        otx_dataset: DatasetEntity,
+        labels: list[LabelEntity],
+        pipeline: Sequence[dict],
+        test_mode: bool = False,
+        max_refetch: int = 1000,
+        **kwargs,
+    ) -> None:
+        """Initialize OTXDetDataset.
+
+        Args:
+            otx_dataset (DatasetEntiy): DatasetEntity from dataset api
+            labels (List[LabelEntity]): List of LabelEntity
+            pipeline (Sequence[dict]): List of data pipeline
+            test_mode (bool): Whether current dataset is for test or not
+            max_refetch (int): If ``Basedataset.prepare_data`` get a None img.
+                The maximum extra number of cycles to get a valid
+            kwargs: Additional kwargs
+        """
+        dataset_cfg = kwargs.copy()
+        _ = dataset_cfg.pop("org_type", None)
+        new_classes = dataset_cfg.pop("new_classes", [])
+        self.otx_dataset = otx_dataset
+        self.labels = labels
+        self.CLASSES = [label.name for label in labels]
```

Comment:
could you ditch `self.CLASSES` member variable? It's a leftover from the old legend MMDet 2.x code.